domain/schema/model.go

```diff
@@ -50,25 +51,148 @@ CREATE TABLE model_config (
 `)
 }
 
-func spacesSchema() schema.Patch {
+func spaceSchema() schema.Patch {
 	return schema.MakePatch(`
-CREATE TABLE provider_spaces (
+CREATE TABLE space (
     uuid            TEXT PRIMARY KEY,
-    name            TEXT
+    name            TEXT NOT NULL,
+    is_public       BOOLEAN
+);
+
+CREATE UNIQUE INDEX idx_spaces_uuid_name
+ON space (name);
+
+CREATE TABLE provider_space (
+    provider_id     TEXT PRIMARY KEY,
+    space_uuid      TEXT NOT NULL,
+    CONSTRAINT      fk_provider_space_space_uuid
+        FOREIGN KEY     (space_uuid)
+        REFERENCES      space(uuid)
 );
 
-CREATE TABLE spaces (
+CREATE UNIQUE INDEX idx_provider_space_space_uuid
+ON provider_space (space_uuid);
+`)
+}
+
+func subnetSchema() schema.Patch {
+	return schema.MakePatch(`
+CREATE TABLE subnet (
+    uuid                         TEXT PRIMARY KEY,
+    cidr                         TEXT NOT NULL,
+    vlan_tag                     INT,
+    is_public                    BOOLEAN,
+    space_uuid                   TEXT,
+    subnet_type_uuid             TEXT,
+    CONSTRAINT                   fk_subnets_spaces
+        FOREIGN KEY                  (space_uuid)
+        REFERENCES                   space(uuid),
+    CONSTRAINT                   fk_subnet_types
+        FOREIGN KEY                  (subnet_type_uuid)
+        REFERENCES                   subnet_type(uuid)
+);
+
+CREATE TABLE subnet_type (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL,
+    is_usable                    BOOLEAN,
+    is_space_settable            BOOLEAN
+);
+
+INSERT INTO subnet_type VALUES
+    (0, 'base', true, true),    -- The base (or standard) subnet type. If another subnet is an overlay of a base subnet in fan bridging, then the base subnet is the underlay in fan terminology.
+    (1, 'fan_overlay', false, false),    
+    (2, 'fan_overlay_segment', true, true);
+
+CREATE TABLE subnet_association_type (
+    uuid                         TEXT PRIMARY KEY,
+    name                         TEXT NOT NULL
+);
+
+INSERT INTO subnet_association_type VALUES
+    (0, 'overlay_of');    -- The subnet is an overlay of other (an underlay) subnet.
+
+CREATE TABLE subnet_type_association_type (
+    subject_subnet_type_uuid       TEXT PRIMARY KEY,
+    associated_subnet_type_uuid    TEXT NOT NULL,
+    CONSTRAINT                     fk_subject_subnet_type_uuid
+        FOREIGN KEY                    (subject_subnet_type_uuid)
+        REFERENCES                     subnet_type(uuid),
+    CONSTRAINT                     fk_associated_subnet_type_uuid
+        FOREIGN KEY                    (associated_subnet_type_uuid)
+        REFERENCES                     subnet_association_type(uuid)
+);
```

Comment:
This needs the association type

We can populate this with some data too. For example, a fan overlay (subject) may have an association type of overlay_of with a base.